lib/bacnet-asn1.js

```diff
@@ -779,7 +784,14 @@ var decode_read_access_result = module.exports.decode_read_access_result = funct
         var localResult = bacapp_decode_application_data(buffer, offset + len, apduLen + offset - 1, value.objectIdentifier.type, new_entry.propertyIdentifier);
         if (!localResult) return;
         len += localResult.len;
-        localValues.push({value: localResult.value, type: localResult.type});
+        var resObj = {
+          value: localResult.value,
+          type: localResult.type
+        };
+        if (localResult.encoding !== undefined) {
```

Comment:
I thought about wrapping it for two reasons. Architecture and consistency: the high level object so far is agnostic in terms of datatype, meaning all other high level functions don't need to know anything about the structure. This allows cleaner code and simpler refactoring. Furthermore, we don't pollute the API with additional string specific parameters on high level. In terms of consistency, other encodings are already wrapped in objects. Your thought on this?